### PR TITLE
[feature](information_schema) add rowsets table into information_schema in cloud mode 

### LINKED ISSUE #32154: [Enhancement] implement SchemaRowsetsScanner::_get_all_rowsets for cloud mode
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

system table `information_schema.rowsets` does not work in storage&compute separation mode due to SchemaRowsetsScanner::_get_all_rowsets is not implemented.

### Solution

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wyxxxcat**
## Proposed changes

Issue Number: close #32154

only query tablet **in cache** for cloud mode.

```
mysql> CREATE TABLE user_info
    -> (
    ->     id INT,
    ->     name VARCHAR(50),
    ->     age INT
    -> )
    -> DISTRIBUTED BY HASH(id) BUCKETS 10
    -> PROPERTIES (
    -> "replication_allocation" = "tag.location.default: 1",
    -> "storage_vault_name" = "s3_vault"
    -> );

mysql> select * from user_info;
+------+---------+------+
| id   | name    | age  |
+------+---------+------+
|    1 | Alice   |   25 |
|    4 | John    |   29 |
|    3 | Charlie |   35 |
|    2 | Bob     |   30 |
|    5 | Mars    |   17 |
+------+---------+------+
5 rows in set (0.03 sec)

mysql> select * from information_schema.rowsets;
+------------+--------------------------------------------------+-----------+-----------------+-----------------+--------------+---------------+-------------+-----------------+----------------+---------------------+------------------------+----------------+
| BACKEND_ID | ROWSET_ID                                        | TABLET_ID | ROWSET_NUM_ROWS | TXN_ID          | NUM_SEGMENTS | START_VERSION | END_VERSION | INDEX_DISK_SIZE | DATA_DISK_SIZE | CREATION_TIME       | NEWEST_WRITE_TIMESTAMP | SCHEMA_VERSION |
+------------+--------------------------------------------------+-----------+-----------------+-----------------+--------------+---------------+-------------+-----------------+----------------+---------------------+------------------------+----------------+
|      10002 | 020000000000000a7c43ec9040f03f5ebddb5c07e1e26ea2 |     10201 |               1 | 179653390268416 |            1 |             2 |           2 |             152 |            627 | 2024-10-29 21:05:59 | 2024-10-29 21:05:59    |              0 |
|      10002 | 0200000000000000fffffffffffffffefffffffffffee486 |     10201 |               0 |               0 |            0 |             0 |           1 |               0 |              0 | 1970-01-01 08:00:00 | 1970-01-01 07:59:59    |              0 |
|      10002 | 02000000000000097c43ec9040f03f5ebddb5c07e1e26ea2 |     10185 |               1 | 179653390268416 |            1 |             2 |           2 |             153 |            629 | 2024-10-29 21:05:59 | 2024-10-29 21:05:59    |              0 |
|      10002 | 0200000000000000fffffffffe444511ffffffffffffffeb |     10185 |               0 |               0 |            0 |             0 |           1 |               0 |              0 | 1970-01-01 08:00:00 | 1970-01-01 07:59:59    |              0 |
|      10002 | 02000000000000067c43ec9040f03f5ebddb5c07e1e26ea2 |     10183 |               0 | 179653390268416 |            0 |             2 |           2 |               0 |              0 | 2024-10-29 21:05:59 | 2024-10-29 21:05:59    |              0 |
|      10002 | 0200000000000000fffffffffffffffeffffffffffffffe9 |     10183 |               0 |               0 |            0 |             0 |           1 |               0 |              0 | 1970-01-01 08:00:00 | 1970-01-01 07:59:59    |              0 |
|      10002 | 02000000000000087c43ec9040f03f5ebddb5c07e1e26ea2 |     10189 |               0 | 179653390268416 |            0 |             2 |           2 |               0 |              0 | 2024-10-29 21:05:59 | 2024-10-29 21:05:59    |              0 |
|      10002 | 0200000000000000ffffffffffffffefffffffffffffec86 |     10189 |               0 |               0 |            0 |             0 |           1 |               0 |              0 | 1970-01-01 08:00:00 | 1970-01-01 07:59:59    |              0 |
|      10002 | 02000000000000077c43ec9040f03f5ebddb5c07e1e26ea2 |     10199 |               0 | 179653390268416 |            0 |             2 |           2 |               0 |              0 | 2024-10-29 21:05:59 | 2024-10-29 21:05:59    |              0 |
|      10002 | 0200000000000000fffffffffeb04512ffffffffffffffeb |     10199 |               0 |               0 |            0 |             0 |           1 |               0 |              0 | 1970-01-01 08:00:00 | 1970-01-01 07:59:59    |              0 |
|      10002 | 02000000000000057c43ec9040f03f5ebddb5c07e1e26ea2 |     10195 |               1 | 179653390268416 |            1 |             2 |           2 |             155 |            639 | 2024-10-29 21:05:59 | 2024-10-29 21:05:59    |              0 |
|      10002 | 0200000000000000ffffffffffffea72ffffffffffffffc5 |     10195 |               0 |               0 |            0 |             0 |           1 |               0 |              0 | 1970-01-01 08:00:00 | 1970-01-01 07:59:59    |              0 |
|      10002 | 02000000000000017c43ec9040f03f5ebddb5c07e1e26ea2 |     10191 |               1 | 179653390268416 |            1 |             2 |           2 |             152 |            628 | 2024-10-29 21:05:59 | 2024-10-29 21:05:59    |              0 |
|      10002 | 0200000000000000fffffffffffffffcffffffffffffffed |     10191 |               0 |               0 |            0 |             0 |           1 |               0 |              0 | 1970-01-01 08:00:00 | 1970-01-01 07:59:59    |              0 |
|      10002 | 02000000000000027c43ec9040f03f5ebddb5c07e1e26ea2 |     10193 |               1 | 179653390268416 |            1 |             2 |           2 |             151 |            622 | 2024-10-29 21:05:59 | 2024-10-29 21:05:59    |              0 |
|      10002 | 0200000000000000fffffffffffffef1ffffffffffee2929 |     10193 |               0 |               0 |            0 |             0 |           1 |               0 |              0 | 1970-01-01 08:00:00 | 1970-01-01 07:59:59    |              0 |
|      10002 | 02000000000000047c43ec9040f03f5ebddb5c07e1e26ea2 |     10197 |               0 | 179653390268416 |            0 |             2 |           2 |               0 |              0 | 2024-10-29 21:05:59 | 2024-10-29 21:05:59    |              0 |
|      10002 | 0200000000000000ffffffffffffffeafffffffffffffffb |     10197 |               0 |               0 |            0 |             0 |           1 |               0 |              0 | 1970-01-01 08:00:00 | 1970-01-01 07:59:59    |              0 |
|      10002 | 02000000000000037c43ec9040f03f5ebddb5c07e1e26ea2 |     10187 |               0 | 179653390268416 |            0 |             2 |           2 |               0 |              0 | 2024-10-29 21:05:59 | 2024-10-29 21:05:59    |              0 |
|      10002 | 0200000000000000fffffffffffffffffffffffffffffa80 |     10187 |               0 |               0 |            0 |             0 |           1 |               0 |              0 | 1970-01-01 08:00:00 | 1970-01-01 07:59:59    |              0 |
+------------+--------------------------------------------------+-----------+-----------------+-----------------+--------------+---------------+-------------+-----------------+----------------+---------------------+------------------------+----------------+
20 rows in set (0.04 sec)

mysql> show tablet 10201\G;
*************************** 1. row ***************************
       DbName: wyx
    TableName: user_info
PartitionName: user_info
    IndexName: user_info
         DbId: 10102
      TableId: 10161
  PartitionId: 10160
      IndexId: 10162
       IsSync: true
        Order: 9
    QueryHits: 0
    DetailCmd: SHOW PROC '/dbs/10102/10161/partitions/10160/10162/10201';
1 row in set (0.00 sec)

mysql> SHOW PROC '/dbs/10102/10161/partitions/10160/10162/10201'\G;
*************************** 1. row ***************************
          ReplicaId: 10202
          BackendId: 10002
            Version: 1
  LstSuccessVersion: 1
   LstFailedVersion: -1
      LstFailedTime: NULL
         SchemaHash: 1263548704
      LocalDataSize: 627
     RemoteDataSize: 0
           RowCount: 1
              State: NORMAL
              IsBad: false
         IsUserDrop: false
VisibleVersionCount: -1
       VersionCount: -1
           PathHash: -1
               Path: 
            MetaUrl: http://192.168.31.122:8440/api/meta/header/10201
   CompactionStatus: http://192.168.31.122:8440/api/compaction/show?tablet_id=10201
  CooldownReplicaId: -1
     CooldownMetaId: 
          QueryHits: 0
   PrimaryBackendId: 10002
1 row in set (0.00 sec)
```

### CODE REVIEW

**Comment by wyxxxcat** [comment]
run buildall

**Comment by wyxxxcat** [comment]
run buildall

**Comment by wyxxxcat** [comment]
run buildall

**Comment by wyxxxcat** [comment]
run buildall

**Comment by wyxxxcat** [comment]
run cloud_p0

**Comment by wyxxxcat** [comment]
run buildall

**Comment by wyxxxcat** [comment]
run p0

**Comment by wyxxxcat** [comment]
run buildall

File: be/src/cloud/cloud_tablet_mgr.cpp [hunk 1]
```diff
@@ -197,6 +197,11 @@ Result<std::shared_ptr<CloudTablet>> CloudTabletMgr::get_tablet(int64_t tablet_i
     CloudTablet* tablet_raw_ptr = reinterpret_cast<Value*>(_cache->value(handle))->tablet.get();
     auto tablet = std::shared_ptr<CloudTablet>(tablet_raw_ptr,
                                                [this, handle](...) { _cache->release(handle); });
+    Status st = _engine.meta_mgr().sync_tablet_rowsets(tablet.get(), warmup_data);
```
Comment (wyxxxcat) [inline be/src/cloud/cloud_tablet_mgr.cpp:200]: tablet not updated in cache

**Comment by wyxxxcat** [comment]
run buildall

**Comment by wyxxxcat** [comment]
run p0

**Comment by wyxxxcat** [comment]
run cloud_p0

File: be/src/exec/schema_scanner/schema_rowsets_scanner.cpp [hunk 2]
```diff
@@ -78,7 +82,19 @@ Status SchemaRowsetsScanner::start(RuntimeState* state) {
 
 Status SchemaRowsetsScanner::_get_all_rowsets() {
     if (config::is_cloud_mode()) {
-        return Status::NotSupported("SchemaRowsetsScanner::_get_all_rowsets is not implemented");
+        // cloud tablets
+        std::vector<std::weak_ptr<CloudTablet>> tablets =
+                ExecEnv::GetInstance()->storage_engine().to_cloud().tablet_mgr().get_weak_tablets();
```
Comment (gavinchou) [inline be/src/exec/schema_scanner/schema_rowsets_scanner.cpp:87]: NOTICE that, tablets are in LRU cache for cloud mode, which means not all tablets can be seen when we try to `select * from information_schema.rowsets`. 
It should be mentioned here.

File: regression-test/suites/query_p0/system/test_query_sys_scan_rowsets.groovy [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            "replication_num" = "1",
+            "disable_auto_compaction" = "true",
+            "enable_single_replica_compaction"="true"
+        );
+    """
+
+    SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss")
+    def now = sdf.format(new Date()).toString();
+    
+    List<List<Object>> rowsets_table_name_tablets   = sql """ show tablets from ${rowsets_table_name} """
```
Comment (deardeng) [inline regression-test/suites/query_p0/system/test_query_sys_scan_rowsets.groovy:53]: please use sql_return_maparray function

**Review by deardeng — CHANGES_REQUESTED** [review]

File: be/src/exec/schema_scanner/schema_rowsets_scanner.cpp (hunk 2 above)
Comment (wyxxxcat) [reply be/src/exec/schema_scanner/schema_rowsets_scanner.cpp:87]: done

File: regression-test/suites/query_p0/system/test_query_sys_scan_rowsets.groovy (hunk 3 above)
Comment (wyxxxcat) [reply regression-test/suites/query_p0/system/test_query_sys_scan_rowsets.groovy]: done

**Comment by wyxxxcat** [comment]
run buildall

**Comment by wyxxxcat** [comment]
@deardeng @gavinchou plz review

**Review by gavinchou — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by Yukang-Lian — APPROVED** [review]

**Pull request merged**